File: suc/kotlin/027.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "028_PDFAnnotator_en"
private const val UI_TYPE = "Productivity"
private const val STYLE_THEME = "Soft Pastel"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFFA0C4FF)
        val secondary = Color(0xFFB7E4C7)
        val tertiary = Color(0xFFFFDDC1)
        val background = Color(0xFFF8F7FF)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFF0F0F5)
        val outline = Color(0xFFD9D9E3)
        val success = Color(0xFFB4E4C7)
        val warning = Color(0xFFFFDDC1)
        val error = Color(0xFFFFB4AB)
        val onPrimary = Color(0xFF001D3D)
        val onSecondary = Color(0xFF003720)
        val onTertiary = Color(0xFF3E2800)
        val onBackground = Color(0xFF1B1B1F)
        val onSurface = Color(0xFF1B1B1F)
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 45.sp, fontWeight = FontWeight.Bold)
        val headline = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.SemiBold)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal, lineHeight = 20.sp)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.5.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(4.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }

    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
    }

    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 2.dp, 1.dp, 0.08f)
        val level2 = ShadowSpec(3.dp, 4.dp, 2.dp, 0.10f)
        val level3 = ShadowSpec(6.dp, 8.dp, 4.dp, 0.12f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    headlineLarge = AppTokens.TypographyTokens.display,
    headlineSmall = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelSmall = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val pages = (1..12).toList()
    var selectedPage by remember { mutableStateOf(1) }
    var selectedTool by remember { mutableStateOf("pen") }

    Scaffold(
        contentWindowInsets = WindowInsets(0),
        containerColor = AppTokens.Colors.background,
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Document.pdf", style = MaterialTheme.typography.titleMedium) },
                navigationIcon = { IconButton(onClick = {}) { Box(Modifier.size(24.dp).background(AppTokens.Colors.surfaceVariant, CircleShape)) } },
                actions = { 
                    Button(onClick = {}, colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary)) {
                        Text("Share", style = MaterialTheme.typography.labelSmall.copy(color = AppTokens.Colors.onPrimary))
                    }
                    Spacer(modifier = Modifier.width(AppTokens.Spacing.sm))
                 }
            )
        }
    ) { paddingValues ->
        Row(
            Modifier.fillMaxSize().padding(paddingValues).padding(AppTokens.Spacing.md)
        ) {
            LazyVerticalGrid(
                columns = GridCells.Fixed(2),
                modifier = Modifier.width(150.dp).fillMaxHeight(),
                contentPadding = PaddingValues(AppTokens.Spacing.sm),
                horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm),
                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
            ) {
                items(pages) { page ->
                    Card(
                        onClick = { selectedPage = page },
                        modifier = Modifier.aspectRatio(0.7f),
                        shape = AppTokens.Shapes.medium,
                        colors = CardDefaults.cardColors(containerColor = if (selectedPage == page) AppTokens.Colors.primary else AppTokens.Colors.surfaceVariant),
                        border = if (selectedPage == page) BorderStroke(2.dp, AppTokens.Colors.primary) else null
                    ) {
                        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                            Text(page.toString(), style = MaterialTheme.typography.labelSmall, color = if(selectedPage == page) AppTokens.Colors.onPrimary else AppTokens.Colors.onSurface)
                        }
                    }
                }
            }
            
            Spacer(modifier = Modifier.width(AppTokens.Spacing.md))

            Box(Modifier.weight(1f).fillMaxHeight()) {
                Card(
                    modifier = Modifier.fillMaxSize(),
                    shape = AppTokens.Shapes.large,
                    colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                    elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation)
                ) {
                    Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                        Text("Page $selectedPage", style = MaterialTheme.typography.headlineSmall)
                    }
                }

                Column(
                    modifier = Modifier.align(Alignment.CenterStart).padding(AppTokens.Spacing.md)
                        .background(AppTokens.Colors.surface.copy(alpha=0.8f), AppTokens.Shapes.large)
                        .border(1.dp, AppTokens.Colors.outline, AppTokens.Shapes.large)
                        .padding(AppTokens.Spacing.sm),
                    verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                ) {
                    AnnotationToolButton(icon = "P", toolName = "pen", selectedTool = selectedTool, onClick = { selectedTool = "pen" })
                    AnnotationToolButton(icon = "H", toolName = "highlighter", selectedTool = selectedTool, onClick = { selectedTool = "highlighter" })
                    AnnotationToolButton(icon = "T", toolName = "text", selectedTool = selectedTool, onClick = { selectedTool = "text" })
                    AnnotationToolButton(icon = "S", toolName = "shape", selectedTool = selectedTool, onClick = { selectedTool = "shape" })
                }
            }
        }
    }
}

@Composable
fun AnnotationToolButton(icon: String, toolName: String, selectedTool: String, onClick: () -> Unit) {
    val isSelected = toolName == selectedTool
    Button(
        onClick = onClick,
        modifier = Modifier.size(40.dp),
        shape = CircleShape,
        colors = ButtonDefaults.buttonColors(containerColor = if(isSelected) AppTokens.Colors.secondary else AppTokens.Colors.surfaceVariant),
        contentPadding = PaddingValues(0.dp)
    ) {
        Text(icon, fontWeight = FontWeight.Bold, color = if(isSelected) AppTokens.Colors.onSecondary else AppTokens.Colors.onSurface)
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, widthDp = 720, heightDp = 1280)
@Composable
fun DefaultPreview() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/027.swift
import SwiftUI

// MARK: - AppTokens Translation
// This section translates the AppTokens object from Kotlin to Swift.
// It defines custom colors, typography styles, shapes, spacing, and elevation for consistent UI elements.

extension Color {
    /// Initializes a Color from a 32-bit hexadecimal integer, similar to Android's 0xFF format.
    /// - Parameters:
    ///   - hex: The hexadecimal value (e.g., 0xFFA0C4FF).
    ///   - alpha: The alpha component, defaults to 1.0 (fully opaque).
    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0,
            opacity: alpha
        )
    }
}

/// A struct containing all design tokens for the application, mirroring Kotlin's AppTokens.
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0xFFA0C4FF)
        static let secondary = Color(hex: 0xFFB7E4C7)
        static let tertiary = Color(hex: 0xFFFFDDC1)
        static let background = Color(hex: 0xFFF8F7FF)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xFFF0F0F5)
        static let outline = Color(hex: 0xFFD9D9E3)
        static let success = Color(hex: 0xFFB4E4C7)
        static let warning = Color(hex: 0xFFFFDDC1)
        static let error = Color(hex: 0xFFFFB4AB)
        static let onPrimary = Color(hex: 0xFF001D3D)
        static let onSecondary = Color(hex: 0xFF003720)
        static let onTertiary = Color(hex: 0xFF3E2800)
        static let onBackground = Color(hex: 0xFF1B1B1F)
        static let onSurface = Color(hex: 0xFF1B1B1F)
    }

    struct TypographyTokens {
        // SwiftUI's Font.system allows specifying size and weight directly.
        // Line height and letter spacing can be applied as separate modifiers if needed globally,
        // but for this UI, direct font application is sufficient.
        static let display = Font.system(size: 45, weight: .bold)
        static let headline = Font.system(size: 24, weight: .semibold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        static let small = RoundedRectangle(cornerRadius: 4.0)
        static let medium = RoundedRectangle(cornerRadius: 8.0)
        static let large = RoundedRectangle(cornerRadius: 12.0)
        static let circle = Circle()
    }

    struct Spacing {
        // Using CGFloat for Dp values, assuming 1dp = 1pt for direct translation.
        static let xs: CGFloat = 4.0
        static let sm: CGFloat = 8.0
        static let md: CGFloat = 12.0
        static let lg: CGFloat = 16.0
        static let xl: CGFloat = 24.0
    }

    /// Represents a shadow specification, similar to Android's elevation.
    struct ShadowSpec {
        let elevation: CGFloat // Corresponds to SwiftUI's shadow radius
        let radius: CGFloat    // Corresponds to SwiftUI's shadow radius (often combined with elevation)
        let dy: CGFloat        // Corresponds to SwiftUI's shadow y-offset
        let opacity: Double    // Corresponds to SwiftUI's shadow color opacity
    }

    struct ElevationMapping {
        // Mapping Android elevation to SwiftUI shadow properties.
        // 'elevation' from Kotlin is often best represented by SwiftUI's 'radius' for visual softness.
        // 'radius' from Kotlin's ShadowSpec is also used for SwiftUI's 'radius'.
        // 'dy' directly maps to SwiftUI's 'y' offset.
        static let level1 = ShadowSpec(elevation: 1.0, radius: 2.0, dy: 1.0, opacity: 0.08)
        static let level2 = ShadowSpec(elevation: 3.0, radius: 4.0, dy: 2.0, opacity: 0.10)
        static let level3 = ShadowSpec(elevation: 6.0, radius: 8.0, dy: 4.0, opacity: 0.12)
    }
}

// MARK: - AppTheme Equivalent
// In SwiftUI, theming is often applied via ViewModifiers or directly to views.
// This modifier acts as a conceptual wrapper for applying global styles if needed,
// though for this specific UI, most styles are applied directly using AppTokens.
struct AppThemeModifier: ViewModifier {
    func body(content: Content) -> some View {
        content
            // You can apply global environment values here if your theme required them,
            // e.g., .environment(\.font, AppTokens.TypographyTokens.body)
            // For this specific UI, direct application of AppTokens is sufficient and more flexible.
    }
}

extension View {
    /// Applies the custom application theme to the view hierarchy.
    func appTheme() -> some View {
        self.modifier(AppThemeModifier())
    }
}

// MARK: - Reusable Components

/// A reusable button for annotation tools, mirroring the Kotlin `AnnotationToolButton` composable.
struct AnnotationToolButton: View {
    let icon: String
    let toolName: String
    let selectedTool: String
    let onClick: () -> Void

    var isSelected: Bool { toolName == selectedTool }

    var body: some View {
        Button(action: onClick) {
            Text(icon)
                .font(Font.system(size: 17, weight: .bold)) // Font size for the icon, adjusted for visual parity
                .foregroundColor(isSelected ? AppTokens.Colors.onSecondary : AppTokens.Colors.onSurface)
        }
        .frame(width: 40, height: 40) // Corresponds to Modifier.size(40.dp)
        .background(isSelected ? AppTokens.Colors.secondary : AppTokens.Colors.surfaceVariant)
        .clipShape(AppTokens.Shapes.circle) // Corresponds to CircleShape
        .buttonStyle(PlainButtonStyle()) // Removes default SwiftUI button styling (e.g., blue tint, tap effects)
        .contentShape(AppTokens.Shapes.circle) // Ensures the tap area matches the circular shape
    }
}

// MARK: - RootScreen Translation

/// The main UI screen, translating the Kotlin `RootScreen` composable.
struct RootScreen: View {
    @State private var selectedPage: Int = 1
    @State private var selectedTool: String = "pen"

    var body: some View {
        // Equivalent of Scaffold in SwiftUI: a VStack for vertical layout,
        // with the top bar and then the main content.
        VStack(spacing: 0) {
            // Top Bar equivalent (CenterAlignedTopAppBar)
            HStack(spacing: AppTokens.Spacing.sm) {
                // Navigation Icon (IconButton with a Box inside)
                Button(action: {}) {
                    AppTokens.Shapes.circle
                        .fill(AppTokens.Colors.surfaceVariant)
                        .frame(width: 24, height: 24) // Corresponds to Modifier.size(24.dp)
                }
                .buttonStyle(PlainButtonStyle()) // Remove default button styling

                Spacer() // Pushes the title to the center

                Text("Document.pdf")
                    .font(AppTokens.TypographyTokens.title)
                    .foregroundColor(AppTokens.Colors.onBackground)

                Spacer() // Pushes the title to the center

                // Actions (Share Button)
                Button(action: {}) {
                    Text("Share")
                        .font(AppTokens.TypographyTokens.label)
                        .foregroundColor(AppTokens.Colors.onPrimary)
                        .padding(.vertical, AppTokens.Spacing.xs) // Corresponds to PaddingValues(0.dp) and internal text padding
                        .padding(.horizontal, AppTokens.Spacing.sm)
                }
                .background(AppTokens.Colors.primary)
                .clipShape(AppTokens.Shapes.small) // Corresponds to RoundedCornerShape(4.dp)
                .buttonStyle(PlainButtonStyle()) // Remove default button styling
            }
            .padding(AppTokens.Spacing.md) // Padding for the top bar content
            .background(AppTokens.Colors.surface) // Top bar background color
            .frame(height: 56) // Standard AppBar height in Material Design 3 is often 64dp, but 56pt is common in iOS and visually fits.

            // Main content area (Row in Compose)
            HStack(spacing: AppTokens.Spacing.md) { // Corresponds to Modifier.padding(AppTokens.Spacing.md) and Spacer(modifier = Modifier.width(AppTokens.Spacing.md))
                // Left side: LazyVerticalGrid for pages
                LazyVGrid(columns: [GridItem(.flexible()), GridItem(.flexible())], spacing: AppTokens.Spacing.sm) { // GridCells.Fixed(2) with spacedBy
                    ForEach(1...12, id: \.self) { page in
                        Button(action: { selectedPage = page }) {
                            AppTokens.Shapes.medium // Card shape (RoundedCornerShape(8.dp))
                                .fill(selectedPage == page ? AppTokens.Colors.primary : AppTokens.Colors.surfaceVariant)
                                .overlay(
                                    Text(page.description)
                                        .font(AppTokens.TypographyTokens.label)
                                        .foregroundColor(selectedPage == page ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                                )
                                .aspectRatio(0.7, contentMode: .fit) // Corresponds to Modifier.aspectRatio(0.7f)
                                .overlay(
                                    // BorderStroke equivalent
                                    selectedPage == page ?
                                    AppTokens.Shapes.medium.stroke(AppTokens.Colors.primary, lineWidth: 2)
                                    : nil
                                )
                        }
                        .buttonStyle(PlainButtonStyle()) // Remove default button styling
                    }
                }
                .frame(width: 150) // Corresponds to Modifier.width(150.dp)
                .padding(AppTokens.Spacing.sm) // Corresponds to contentPadding for LazyVerticalGrid
                .frame(maxHeight: .infinity) // Corresponds to .fillMaxHeight()

                // Right side: Main document view (Box in Compose)
                ZStack(alignment: .center) { // Corresponds to Box(contentAlignment = Alignment.Center)
                    // Main document card
                    AppTokens.Shapes.large // Card shape (RoundedCornerShape(12.dp))
                        .fill(AppTokens.Colors.surface)
                        .shadow(color: Color.black.opacity(AppTokens.ElevationMapping.level1.opacity),
                                radius: AppTokens.ElevationMapping.level1.radius, // Using radius for softness
                                x: 0,
                                y: AppTokens.ElevationMapping.level1.dy) // Shadow for elevation
                        .overlay(
                            Text("Page \(selectedPage)")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)
                        )
                        .frame(maxWidth: .infinity, maxHeight: .infinity) // Corresponds to Modifier.fillMaxSize()

                    // Annotation tools column
                    VStack(spacing: AppTokens.Spacing.sm) { // Corresponds to verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                        AnnotationToolButton(icon: "P", toolName: "pen", selectedTool: selectedTool, onClick: { selectedTool = "pen" })
                        AnnotationToolButton(icon: "H", toolName: "highlighter", selectedTool: selectedTool, onClick: { selectedTool = "highlighter" })
                        AnnotationToolButton(icon: "T", toolName: "text", selectedTool: selectedTool, onClick: { selectedTool = "text" })
                        AnnotationToolButton(icon: "S", toolName: "shape", selectedTool: selectedTool, onClick: { selectedTool = "shape" })
                    }
                    .padding(AppTokens.Spacing.sm) // Corresponds to .padding(AppTokens.Spacing.sm)
                    .background(AppTokens.Colors.surface.opacity(0.8)) // Corresponds to .background(AppTokens.Colors.surface.copy(alpha=0.8f))
                    .clipShape(AppTokens.Shapes.large) // Corresponds to AppTokens.Shapes.large
                    .overlay(
                        AppTokens.Shapes.large // Border equivalent
                            .stroke(AppTokens.Colors.outline, lineWidth: 1) // Corresponds to .border(1.dp, AppTokens.Colors.outline, AppTokens.Shapes.large)
                    )
                    .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading) // Corresponds to .align(Alignment.CenterStart)
                    .padding(AppTokens.Spacing.md) // Corresponds to .padding(AppTokens.Spacing.md)
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity) // Corresponds to Modifier.weight(1f).fillMaxHeight()
            }
            .padding(.horizontal, AppTokens.Spacing.md) // Overall padding for the main content row (left/right)
            .frame(maxWidth: .infinity, maxHeight: .infinity) // Ensure main content fills remaining space
        }
        .background(AppTokens.Colors.background.ignoresSafeArea()) // Apply background to the whole screen, ignoring safe area
        .statusBarHidden(true) // Hide the status bar for a full-screen effect (iOS 14+)
    }
}

// MARK: - SwiftUI App Entry Point

/// The main entry point for the SwiftUI application, mirroring the `MainActivity`'s `onCreate` setup.
@main
struct SingleFileUIApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
                .appTheme() // Apply the custom theme modifier
        }
    }
}